src/js/authEthereum.js

```diff
@@ -42,17 +42,23 @@ async function login () {
     render()
   })
   provider.on('chainChanged', (chainId) => {
-    window.connectedEthNetwork = ethNetworks[chainId]
+    window.connectedEthNetwork = chainIdToEthNetwork[chainId]
+    window.isValidEthNetwork = window.connectedEthNetwork === process.env.ethNetworkId
     render()
   })
 
   window.ethUserAddress = provider.selectedAddress
-  window.connectedEthNetwork = ethNetworks[provider.chainId]
+  window.connectedEthNetwork = chainIdToEthNetwork[provider.chainId]
+  window.isValidEthNetwork = window.connectedEthNetwork === process.env.ethNetworkId
   window.ethInitialized = true
 
   render()
 
-  if (window.nearInitialized) checkTransferStatuses({ loop: window.LOOP_INTERVAL })
+  if (window.nearInitialized) {
+    checkTransferStatuses(
+      { loop: window.LOOP_INTERVAL, expectedEthChainId: ethNetworkToChainId[process.env.ethNetworkId] }
```

Comment:
Can we get rid of `expectedEthChainId` here and add `checkTransferStatuses` call to `chainChanged`? Then everywhere that we call `checkTransferStatuses` we need to add `window.isValidEthNetwork` check before calling `checkTransferStatuses`.

Reply:
But checkTransferStatuses is called once here and keeps looping in the background so it does need to check that the chain id is correct.

Reply:
But you don't need to start looping it in the background here.

I guess there's this scenario:

1. I have correct Eth network selected
2. I log in with both networks
3. I select a different Eth network
4. `checkStatusAll` keeps looping, but shouldn't

For that, I'd rather see an implementation like this:

```js
if (window.nearInitialized && window.isValidEthNetwork) {
  window.stopCheckingStatuses = checkStatusAll({ loop: window.LOOP_INTERVAL })
}
provider.on('chainChanged', (chainId) => {
  window.connectedEthNetwork = ethNetworks[chainId]
  window.connectedEthNetwork = chainIdToEthNetwork[chainId]
  window.isValidEthNetwork = window.connectedEthNetwork === process.env.ethNetworkId
  if (!window.isValidEthNetwork) {
    window.stopCheckingStatuses()
  } else {
    if (window.stopCheckingStatuses) window.stopCheckingStatuses()
    window.stopCheckingStatuses = checkStatusAll({ loop: window.LOOP_INTERVAL })
  }
  render()
})
```

Reply:
That code could probably be cleaned up a bit, but basically I think adding `expectedEthChainId` to `checkStatusAll` is pretty ugly, and I'd rather see it return a function that allows you to stop looping.

Reply:
After discussion, `stopCheckStatuses()` and `expectedEthChainId` would not solve the issue properly and the network check should be done at checkSync / checkUnlock / checkApprove / checkLock to be sure that the wrong eth network is not used.
This PR now only prevents rendering if `!window.isValidEthNetwork` and the checkStatus issue is fixed in the lib https://github.com/near/rainbow-bridge-client/pull/4